cirq/sim/clifford/clifford_simulator.py

```diff
@@ -300,25 +316,55 @@ def apply_unitary(self, op: 'cirq.Operation'):
                              self.qubit_map[op.qubits[1]])
             self.ch_form._CZ(self.qubit_map[op.qubits[0]],
                              self.qubit_map[op.qubits[1]])
-        elif op.gate == cirq.Z:
-            self.tableau._Z(self.qubit_map[op.qubits[0]])
-            self.ch_form._Z(self.qubit_map[op.qubits[0]])
-        elif op.gate == cirq.X:
-            self.tableau._X(self.qubit_map[op.qubits[0]])
-            self.ch_form._X(self.qubit_map[op.qubits[0]])
-        elif op.gate == cirq.Y:
-            self.tableau._Y(self.qubit_map[op.qubits[0]])
-            self.ch_form._Y(self.qubit_map[op.qubits[0]])
-        elif op.gate == cirq.S:
-            self.tableau._S(self.qubit_map[op.qubits[0]])
-            self.ch_form._S(self.qubit_map[op.qubits[0]])
-        elif op.gate == cirq.H:
-            self.tableau._H(self.qubit_map[op.qubits[0]])
-            self.ch_form._H(self.qubit_map[op.qubits[0]])
         else:
-            raise ValueError('%s cannot be run with Clifford simulator' %
+            raise ValueError('%s cannot be run with Clifford simulator.' %
                              str(op.gate))  # type: ignore
 
+    def apply_single_qubit_unitary(self, op: 'cirq.Operation'):
+        assert op.gate is not None
+        qubit = self.qubit_map[op.qubits[0]]
+        if op.gate == cirq.H:
```

Comment:
It would be nice to have a comment explaining why we did this special handling so that future readers aren't confused why `S` isn't included here or if `H` cannot be handled by the process below

Reply:
That's why I didn't want to have any special case handling in first place...

Reply:
Added a comment.